executor/load_data.go

```diff
@@ -634,29 +696,81 @@ func (e *LoadDataWorker) Load(ctx context.Context, reader io.ReadSeekCloser) err
 	}
 	parser.SetLogger(log.Logger{Logger: logutil.Logger(ctx)})
 
-	err = sessiontxn.NewTxn(ctx, e.Ctx)
+	progress := asyncloaddata.Progress{
+		SourceFileSize: -1,
+		LoadedFileSize: 0,
+		LoadedRows:     0,
+	}
+	// TODO try to read total file size from remote storage. gcs seems wrongly
+	// implement io.Seek
+	e.progress.Store(&progress)
+
+	err = asyncloaddata.StartJob(ctx, sqlExec, jobID)
 	if err != nil {
 		return err
 	}
+
+	failpoint.Inject("AfterStartJob", nil)
+
 	group, groupCtx := errgroup.WithContext(ctx)
+	// done is used to let commitWork goroutine notify UpdateJobProgress
+	// goroutine that the job is finished.
+	done := make(chan struct{})
 
-	// processStream process input data, enqueue commit task
+	// UpdateJobProgress goroutine.
 	group.Go(func() error {
-		return e.processStream(groupCtx, parser)
+		ticker := time.NewTicker(time.Duration(asyncloaddata.HeartBeatInSec) * time.Second)
+		defer ticker.Stop()
+
+		for {
+			select {
+			case <-done:
+				// try to update progress to set 100% progress
+				p := e.progress.Load()
+				ok, err2 := asyncloaddata.UpdateJobProgress(ctx, sqlExec, jobID, p.String())
+				if !ok || err2 != nil {
+					logutil.Logger(ctx).Warn("failed to update job progress when finished",
+						zap.Bool("ok", ok), zap.Error(err2))
```

Comment:
handle error?

Reply:
here the whole job is finished, only updating the loaded_file_size is failed. I don't want to let it affect the main process.